network/p2p/blob/blob_service.go

```diff
@@ -300,6 +302,45 @@ func AuthorizedRequester(
 	}
 }
 
+// RateLimitingFilter returns a PeerBlockRequestFilter function used by bitswap filter blob requests.
+// The callback returns true if the peer is NOT rate limited.
+func RateLimitingFilter(
+	logger zerolog.Logger,
+	limit float64,
+	lockout time.Duration,
+) func(peer.ID, cid.Cid) bool {
+
+	// rate limiter to use for all incoming blob requests.
+	// Note: burst is hardcoded to 1 since this callback handles a single CID at a time.
+	limiter := ratelimiter.NewRateLimiter(rate.Limit(limit), 1, lockout)
```

Comment:
this may not be appropriate for use on the public network since peerIDs can be changed arbitrarily